src/coreclr/jit/fgdiagnostic.cpp

```diff
@@ -4200,6 +4200,76 @@ class SsaCheckVisitor : public GenTreeVisitor<SsaCheckVisitor>
     void SetBlock(BasicBlock* block)
     {
         m_block = block;
+        CheckPhis(block);
+    }
+
+    void CheckPhis(BasicBlock* block)
+    {
+        for (Statement* const stmt : block->Statements())
+        {
+            if (!stmt->IsPhiDefnStmt())
+            {
+                break;
+            }
```

Comment:
We already have NonPhiStatements(); maybe we should have PhiStatements()? (Presumably somebody else would verify all the PHI statements appear first).

```suggestion
        for (Statement* const stmt : block->PhiStatements())
        {
```

Reply:
I should probably check that all the PhiDefs appear first, but then I wouldn't need the partial iterator.